src/flow/nav/FlowStackRoutes.ts

```diff
@@ -26,6 +28,45 @@ export const FLOW_STACK_SCREEN_ROUTES = Object.keys(FLOW_STACK_SCREENS) as FlowS
 
 export const INITIAL_FLOW_STACK_SCREEN_ROUTE = HomeScreen.ROUTE;
 
+export const FLOW_STACK_CONFIGURATIONS: {
+  initialRouteNames: FlowStackRoute[];
+  preInitialRouteName: FlowStackRoute;
+} = {
+  initialRouteNames: [LoginScreen.ROUTE, RegisterScreen.ROUTE],
+  preInitialRouteName: LoadingScreen.ROUTE,
+};
+
+export const routeIsFlowStackInitialRouteName = (routeName: string | null): routeName is FlowStackRoute => {
+  return FLOW_STACK_CONFIGURATIONS.initialRouteNames.includes(routeName as FlowStackRoute);
+};
+
+export const routeIsFlowStackPreInitialRouteName = (routeName: string | null): routeName is FlowStackRoute => {
+  return routeName === FLOW_STACK_CONFIGURATIONS.preInitialRouteName;
+};
+
+export const getFlowStackScreenRoutes = (initialRouteName: FlowStackRoute | null) => {
+  if (routeIsFlowStackInitialRouteName(initialRouteName)) {
+    const {
+      /* eslint-disable @typescript-eslint/no-unused-vars */
+      [initialRouteName]: _,
+      [FLOW_STACK_CONFIGURATIONS.preInitialRouteName]: __,
+      /* eslint-enable @typescript-eslint/no-unused-vars */
+      ...screens
+    } = FLOW_STACK_SCREENS;
+
+    const routes: FlowStackRoute[] = [initialRouteName].concat(Object.keys(screens) as FlowStackRoute[]);
+
+    return routes;
+  } else {
+    const routes: FlowStackRoute[] = [
+      // multiline
+      LoadingScreen.ROUTE,
+    ];
+
+    return routes;
+  }
+};
```

Comment:
[3/3] 

The `initialRouteName` determines which screens are queued up in the `<FlowStack.Navigator>` component. Since `initialRouteName` can't be set (the `<FlowStack.Navigator>` component isn't remounted in this approach), it is not used. Instead, the `<FlowStack.Navigator>` component determines the initial route by selecting the first element in the array of navigator screen children rendered in [1/3].

| `initialRouteName` | `routes` (stack) |
| --- | --- |
| `null` | `[ "Loading" ]` |
| `"Login"` | `[ "Login", "Home", "Logout", "Register", ... ]`, where `"Loading"` has been removed |
| `"Register"` | `[ "Register", "Home", "Register", "Logout", ... ]`, where `"Loading"` has been removed |